pkg/virt-api/rest/subresource.go

```diff
@@ -246,8 +246,8 @@ func (app *SubresourceAPIApp) RestartVMRequestHandler(request *restful.Request,
 			response.WriteError(http.StatusInternalServerError, err)
 			return
 		}
-		// If there's no VMI, just request to start
-		startOnly = true
+		response.WriteError(http.StatusBadRequest, fmt.Errorf("VM is not running"))
```

Comment:
This is entirely my opinion, but I'm not 100% sure what to do here. In the old system v init sense of the word, saying "restart" means "just make it run", but on the other hand if you say "restart means stop, then start"--trying to stop a stopped VM would be an error. @rmohr @davidvossel opinions?

However, if we do decide this is the way to go, `startOnly` is no longer used and needs to be removed.

Reply:
Removed `startOnly`.

Reply:
@davidvossel I'm guessing that you don't have a different opinion? So then I will resolve this discussion.

Reply:
sorry i'm late to this one. 

Here's my thought process...

If we fail to start and already started VM, and we fail to stop and already stopped VM, then we should fail to restart a VM that isn't running.